lib/withCache.ts

```diff
@@ -0,0 +1,49 @@
+import { NextApiRequest, NextApiResponse } from 'next';
+import { CronJob } from 'cron';
+import cache from './cache';
+import { fetchHakkasanByRef } from './hakkasan';
+import type { HakkasanByRefResponse } from './hakkasan';
+import type { Cache } from './cache';
+
+export interface HakkasanRequest extends NextApiRequest {
+  query: {
+    ref: string | string[];
+  };
+  cache: Cache<HakkasanByRefResponse>;
+}
+
+type Handler = (req: HakkasanRequest, res: NextApiResponse) => Promise<void>;
+
+const cron = new CronJob(
+  '00 00 00 * * *',
```

Comment:
Nit: We can just use single `0` values, I've never actually seen double zeroes used before :p

Reply:
Omg whaaaaat - thank you!

Reply:
That's strange 🤔 I wonder why that is. From the man page of crontab(5):
```
          field          allowed values
          -----          --------------
          minute         0-59
          hour           0-23
          day of month   1-31
          month          1-12 (or names, see below)
          day of week    0-7 (0 or 7 is Sun, or use names)
```
I wonder if it has to do with their non-standard seconds value? I'll leave it to you whether to update it or leave it, as I'm not familiar with `node-cron` 👌